extensions/arc/src/ui/dashboards/postgres/postgresComputeAndStoragePage.ts

```diff
@@ -0,0 +1,518 @@
+			// to clear all inputs and return false
+			this.discardButton!.enabled = true;
+			return false;
+		} else {
+			// if a valid value has been entered into the input box, enable save and discard buttons
+			// so that user could choose to either edit instance or clear all inputs
+			// return true
+			this.saveButton!.enabled = true;
+			this.discardButton!.enabled = true;
+			return true;
+		}
+
+	}
+
+	private editWorkerNodeCount() {
+		let currentShards = this._postgresModel.config?.spec.scale.shards;
+
+		this.workerBox!.min = currentShards;
+		this.workerBox!.placeHolder = currentShards!.toString();
+		this.workerBox!.value = '';
+
+		this.saveArgs.workers = undefined;
+	}
+
+	private createCoresMemorySection(): azdata.DivContainer {
+		const titleFlex = { flex: `0 1 250px` };
+
+		const flexContainer = this.modelView.modelBuilder.flexContainer().withLayout({
+			flexWrap: 'wrap',
+			alignItems: 'center'
+		}).component();
+
+		const titleComponent = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.configurationPerNode,
+			CSSStyles: { ...cssStyles.title, 'font-weight': 'bold', 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const titleContainer = this.modelView.modelBuilder.flexContainer().withLayout({ alignItems: 'center' }).component();
+		titleContainer.addItem(titleComponent, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '15px' } });
+
+		const information = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			iconPath: IconPathHelper.information,
+			title: loc.configurationInformation,
+			width: '12px',
+			height: '12px',
+			enabled: false
+		}).component();
+
+		titleContainer.addItem(information, { CSSStyles: { 'margin-left': '5px', 'margin-bottom': '15px' } });
+		flexContainer.addItem(titleContainer, titleFlex);
+
+		let configurationSection = this.modelView.modelBuilder.divContainer().component();
+		configurationSection.addItem(flexContainer);
+
+		return configurationSection;
+	}
+
+	private editCores(type: string) {
+		let currentCPUSize: string | undefined;
+		if (type === loc.coresRequest) {
+			currentCPUSize = this._postgresModel.config?.spec.scheduling?.default?.resources?.requests?.cpu;
+
+			if (!currentCPUSize) {
+				currentCPUSize = '';
+			}
+
+			this.coresRequestBox!.placeHolder = currentCPUSize;
+			this.coresRequestBox!.value = '';
+
+			this.saveArgs.coresRequest = undefined;
+
+		} else if (type === loc.coresLimit) {
+			currentCPUSize = this._postgresModel.config?.spec.scheduling?.default?.resources?.limits?.cpu;
+
+			if (!currentCPUSize) {
+				currentCPUSize = '';
+			}
+
+			this.coreLimitBox!.placeHolder = currentCPUSize;
+			this.coreLimitBox!.value = '';
+
+			this.saveArgs.coresLimit = undefined;
+
+		}
+	}
+
+	private editMemory(type: string) {
+		let currentMemorySize: string | undefined;
+		let currentMemSizeConversion: string;
+
+		if (type === loc.memoryRequest) {
```

Comment:
I'd suggest making the logic all the same and just at the beginning initialize the currentMemorySize and which components are updated in the `if (type === ...` statement. Bit cleaner and less chance of errors if we mess updating one up.

Reply:
Same for the editCores above.